imports/plugins/core/accounts/server/init.js

```diff
@@ -163,25 +143,34 @@ Meteor.startup(() => {
       emailIsVerified = true;
     }
 
-    // clone before adding roles
-    const account = Object.assign({ shopId }, user, additionals);
-    account.userId = user._id;
-    Collections.Accounts.insert(account);
-
-    const insertedAccount = Collections.Accounts.findOne({ userId: user._id });
-    Promise.await(appEvents.emit("afterAccountCreate", {
-      account: insertedAccount,
-      createdBy: user._id
-    }));
-
-    // send a welcome email to new users,
+    // create a tokenObj and send a welcome email to new users,
     // but skip the first default admin user and anonymous users
     // (default admins already get a verification email)
+    let tokenObj;
     if (shopId && !emailIsVerified && user.emails[0]) {
-      const tokenObj = generateVerificationTokenObject({ address: user.emails[0].address });
+      tokenObj = generateVerificationTokenObject({ address: user.emails[0].address });
+    }
+
+    // Get GraphQL context to pass to mutation
+    // This is the only place in the app that still
+    // uses `getGraphQLContextInMeteorMethod`
+    // Prioritize removing if possible
+    const context = Promise.await(getGraphQLContextInMeteorMethod(null));
+
+    context.isInternalCall = true;
+
+    Promise.await(context.mutations.createAccount(context, {
+      additionals,
+      groupId: groupToAddUser,
+      shopId,
+      tokenObj,
+      user
```

Comment:
Let's take this opportunity to flatten this input. There aren't that many actual fields we need, so having the sub-objects of `additionals` and `user` seems overly complex. I suggest:

```js
{
  shopId,
  name,
  picture,
  bio,
  username,
  userId,
  verificationToken
}
```

And update GraphQL input schema types to match that.

I think that's all the info createAccount needs. Note the following:
- Move the whole user.services block back into here, and use it to set `name`, `picture`, `bio`, and `username` if possible. Then pass just those to createAccount.
- Instead of passing the whole tokenObj, pass just `verificationToken: tokenObj.token`. I don't think it needs any more token info.
- I don't think we should support passing in a `groupId`. It would need security checks to be safe. Instead, can we replace all the groups lookup in both functions with `const group = await Groups.findOne({ slug: "customer", shopId });`? In other words, put all new accounts in the "customer" group, and they can call the other mutation to set a new group after. The only place we pass `groupId` is in `inviteShopMember`, so you can update that function to call `addAccountToGroup` after creating the user and get the same effect.

Reply:
There is also the `profile` object on `createUser`, which has `invited` on it at the moment. Should this be passed to `createAccount` as well?

Reply:
> I don't think we should support passing in a groupId. It would need security checks to be safe. Instead, can we replace all the groups lookup in both functions with const group = await Groups.findOne({ slug: "customer", shopId });? In other words, put all new accounts in the "customer" group, and they can call the other mutation to set a new group after. The only place we pass groupId is in inviteShopMember, so you can update that function to call addAccountToGroup after creating the user and get the same effect.

It looks like we might already be doing this: https://github.com/reactioncommerce/reaction/blob/refactor-kieckhafer-demeteorize-SendWelcomeEmail/imports/plugins/core/accounts/server/no-meteor/mutations/inviteShopMember.js#L125

Am I missing something, or can I just remove the group code without adding something back?

Reply:
It does look like we're doing it doubly. Seems fine to just remove groupId.